integrationtest/hostmetrics_test.go

```diff
@@ -120,7 +133,9 @@ func loadExpectedMetrics(t *testing.T, expectedMetricsPath string) map[string]Ex
 		if _, ok := result[expect.Name]; ok {
 			t.Fatalf("Found multiple ExpectedMetric entries with Name=%q", expect.Name)
 		}
-		result[expect.Name] = expect
+		if len(expect.OnlyOn) == 0 || sliceContains(expect.OnlyOn, runtime.GOOS) {
```

Comment:
If `only_on` isn't in the yaml block, is it `nil` or an empty slice? I thought it was `nil`, but since this works maybe that's not the case.

Reply:
len(nil) is zero. There is an internal style rule that apis should treat nil and an empty slice the same: go/go-style/decisions#nil-slices, and len() and probably other parts of the standard library adhere to this rule.